traits/trait_base.py

```diff
@@ -205,6 +205,11 @@ def safe_contains(value, container):
     TypeError.  In these cases we make the (reasonable) assumption that the
     value is _not_ contained in the container.
     """
+    # Do a LBYL check for Enums, to avoid the DeprecationWarning issued
+    # by Python 3.7. Ref: enthought/traits#853.
+    if isinstance(container, enum.EnumMeta) and not isinstance(value, enum.Enum):
```

Comment:
I'm not hugely happy with this solution. An alternative is to use the `filterwarnings` machinery to explicitly filter out the relevant `DeprecationWarning`, but that's awkward, too: either you do it globally, in which case you're violating DNMWGSTDNBTY, or you have the performance hit of saving and restoring warnings state on every call to `safe_contains`.

Reply:
What happens if you have an `enum.Enum` instance that is not an enum from `container`?  Do we need `not isinstance(value, container)` instead?

Reply:
I think that's covered by the test suite.

Reply:
Actually, I'm not sure I understand what you're asking. Are you asking about the case that `value` is an `enum.Enum` instance and `container` is a different `enum.Enum` class?

Reply:
Yes - question resolved in offline conversation.